cpp/examples/ws-server/src/main.cpp

```diff
@@ -57,7 +57,12 @@ int main(int argc, const char* argv[]) {
   options.callbacks.onClientUnadvertise = [](uint32_t clientId, uint32_t clientChannelId) {
     std::cerr << "Client " << clientId << " unadvertised channel " << clientChannelId << '\n';
   };
-  foxglove::WebSocketServer server{options};
+  auto serverResult = foxglove::WebSocketServer::create(std::move(options));
```

Comment:
This seems to be a canonical way to create class instances if the constructor can fail and return an expected result. But it's new to me, so I can't be certain.